core/src/main/java/bio/terra/pearl/core/service/workflow/EnrollmentService.java

```diff
@@ -136,20 +141,32 @@ public HubResponse<Enrollee> enroll(String portalShortcode, EnvironmentName envN
      * the "enrollee"
      */
     @Transactional
-    public HubResponse<Enrollee> enrollAsProxy(String portalShortcode, EnvironmentName envName, String studyShortcode, ParticipantUser user, PortalParticipantUser ppUser, UUID preEnrollResponseId) {
-        HubResponse<Enrollee> proxyResponse = enroll(portalShortcode, envName, studyShortcode, user, ppUser, preEnrollResponseId, false);
-        HubResponse<Enrollee> governedResponse = enrollGovernedUser(portalShortcode, envName, studyShortcode, proxyResponse.getEnrollee(), user, ppUser, preEnrollResponseId);
-        governedResponse.setEnrollee(proxyResponse.getEnrollee());
+    public HubResponse<Enrollee> enrollAsProxy(EnvironmentName envName, String studyShortcode, ParticipantUser proxyUser,
+                                               PortalParticipantUser ppUser, UUID preEnrollResponseId) {
+        Optional<Enrollee> maybeProxyEnrollee =
+                enrolleeService.findOneByParticipantUserIdAndStudyEnvironmentId(proxyUser.getId(), studyShortcode, envName);
+        if (maybeProxyEnrollee.isEmpty()) {
+            maybeProxyEnrollee = Optional.of(enroll(envName, studyShortcode, proxyUser, ppUser, null, false).getEnrollee());
+        }
+        Enrollee proxyEnrollee = maybeProxyEnrollee.orElseThrow(() -> new NotFoundException(
+                "Proxy enrolee with participant user id %s for study %s env %s was not found".formatted(proxyUser.getId(), studyShortcode,
```

Comment:
this saves some extra optional wrapping
```suggestion
        Enrollee proxyEnrollee =
                enrolleeService.findOneByParticipantUserIdAndStudyEnvironmentId(proxyUser.getId(), studyShortcode, envName).orElse(() -> enroll(envName, studyShortcode, proxyUser, ppUser, null, false).getEnrollee());
```